File: src/main/java/ru/spbstu/spartamonitor/logger/Logger.java
package ru.spbstu.spartamonitor.logger;

import java.util.HashMap;

import static ru.spbstu.spartamonitor.config.Config.debugDrawing;

public class Logger {

    private static final HashMap<String, Long> timers = new HashMap<>();

    public static void startTimer(String timerName) {
        if (debugDrawing) {
            //System.out.printf("Start execution for '%s'\n", timerName);
        }
        timers.put(timerName, System.currentTimeMillis());
    }

    public static float releaseTimer(String timerName) {
        if (debugDrawing) {
            System.out.printf("Execution time for '%s' is %.2f s\n", timerName,
                    (float) (System.currentTimeMillis() - timers.get(timerName)) / 1000);
        }
        return System.currentTimeMillis() - timers.get(timerName);
    }
}

File: src/main/kotlin/ru/spbstu/spartamonitor/logger/Logger.kt
package ru.spbstu.spartamonitor.logger

import config.DEBUG_DRAWING

object Logger {
    private val timers = mutableMapOf<String, Long>()

    @JvmStatic
    fun startTimer(timerName: String) {
//        if (Config.debugDrawing) {
//            println("Start execution for '$timerName'");
//        }
        timers[timerName] = System.currentTimeMillis()
    }

    @JvmStatic
    fun releaseTimer(timerName: String): Float {
        if (DEBUG_DRAWING) {
            System.out.printf(
                "Execution time for '$timerName' is %.2f s\n",
                (System.currentTimeMillis() - timers[timerName]!!).toFloat() / 1000
            )
        }
        return (System.currentTimeMillis() - timers[timerName]!!).toFloat()
    }
}
